### PR TITLE
[DCP] Adds utility for converting torch save to dcp

### PR DESCRIPTION
**Pull request by LucasLLC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119335
* __->__ #119333
* #118894
* #119332

[DCP] Adds utility for converting torch save to dcp

Differential Revision: [D53497640](https://our.internmc.facebook.com/intern/diff/D53497640/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

File: torch/distributed/checkpoint/format_utils.py [hunk 1]
```diff
@@ -81,3 +76,21 @@ def dcp_to_torch_save(
         no_dist=True,
     )
     torch.save(sd, torch_save_fn)
+
+
+def torch_save_to_dcp(
+    torch_save_fn: Union[str, os.PathLike], dcp_checkpoint_dir: Union[str, os.PathLike]
```
Comment (fegin) [inline torch/distributed/checkpoint/format_utils.py:82]: Same as the previous PR, we should not use `fn` to represent file name.

File: torch/distributed/checkpoint/format_utils.py [hunk 2]
```diff
@@ -81,3 +76,21 @@ def dcp_to_torch_save(
...
+    Args:
+        torch_save_fn: Filename to store the converted Torch save file.
+        dcp_checkpoint_dir: Directory containing the DCP checkpoint.
+
+    .. warning::
+        To avoid OOM, it's recommended to only run this function on a single rank.
+    """
+
+    state_dict = torch.load(torch_save_fn)
+    dcp.save(state_dict, checkpoint_id=dcp_checkpoint_dir, no_dist=True)
```
Comment (fegin) [inline torch/distributed/checkpoint/format_utils.py:96]: Can we also add main so that users can directly use this file to do conversion? We can do this in a separate PR.

**Comment by LucasLLC** [comment]
Closed in favor of https://github.com/pytorch/pytorch/pull/119816 due to CI issues
